packages/client/src/runtime/core/init/getEngineInstance.ts

```diff
@@ -8,35 +9,40 @@ import { resolveDatasourceUrl } from './resolveDatasourceUrl'
  * Get the engine instance based on the engine type and the target engine type
  * (binary, library, data proxy). If the URL is a prisma:// URL, it will always
  * use the DataProxyEngine. Basically decides which engine to load.
- * @param _engineConfig
+ * @param clientConfig
+ * @param engineConfig
  * @returns
  */
-export function getEngineInstance(_engineConfig: EngineConfig) {
+export function getEngineInstance(clientConfig: GetPrismaClientConfig, engineConfig: EngineConfig) {
   let url: string | undefined
 
   try {
     url = resolveDatasourceUrl({
-      inlineDatasources: _engineConfig.inlineDatasources,
-      overrideDatasources: _engineConfig.overrideDatasources,
-      env: { ..._engineConfig.env, ...process.env },
-      clientVersion: _engineConfig.clientVersion,
+      inlineDatasources: engineConfig.inlineDatasources,
+      overrideDatasources: engineConfig.overrideDatasources,
+      env: { ...engineConfig.env, ...process.env },
+      clientVersion: engineConfig.clientVersion,
     })
   } catch {
     // the error does not matter, but that means we don't have a valid url which
     // means we can't use the DataProxyEngine and will default to LibraryEngine
   }
 
-  const engineType = getClientEngineType(_engineConfig.generator!)
+  if (clientConfig.noEngine !== true && url?.startsWith('prisma://')) {
+    console.warn('In production, we recommend using `prisma generate --no-engine` (See: `prisma generate --help`)')
```

Comment:
We end up in a similar case than `--data-proxy` here? We would need an env var for some cases for this to work?

Reply:
Good point, I'll add it.